tools/calyx-pass/src/main.rs

```diff
@@ -0,0 +1,100 @@
+use calyx_pass::{
+    cli::ParseArgs,
+    pass_explorer::{Breakpoint, PassExplorer},
+    tui::PassExplorerTUI,
+    util,
+};
+use crossterm::style::Stylize;
+use std::{path::PathBuf, process::exit};
+use tempdir::TempDir;
+
+fn fail<F>(message: &str, explain: F)
+where
+    F: FnOnce(),
+{
+    println!("{}: {}", "error".red().bold(), message);
+    println!();
+    explain();
+    println!();
+    println!("Run calyx-pass --help for more information.");
+    exit(1);
+}
+
+#[allow(clippy::write_literal)]
+#[allow(clippy::needless_range_loop)]
+fn main() -> std::io::Result<()> {
+    let mut args: ParseArgs = argh::from_env();
+
+    if args.version {
+        println!("calyx-pass v{}", env!("CARGO_PKG_VERSION"));
+        println!();
+        println!("Features (v{}):", env!("CARGO_PKG_VERSION"));
+        println!(" - View pass transformations");
+        println!(" - Apply or skip passes");
+        println!(" - Focus a component");
+        println!(" - Set breakpoints");
+        println!();
+        println!("See more at https://github.com/calyxir/calyx/blob/calyx-pass/tools/calyx-pass/README.md");
+        return Ok(());
+    }
+
+    // If the user provided no --calyx-exec (or passed in an empty string),
+    // then we first try to obtain the location via fud, and otherwise default
+    // to target/debug/calyx
+    if args.calyx_exec.is_empty() {
```

Comment:
This is tool architecture question: why do we use an external `calyx` binary instead of just bundling the `calyx_opt` crate and using it to drive the tool? An obvious answer could be "so that we don't have to rebuilt the tool every time we rebuild the compiler".

Reply:
Yes